src/AppInstallerCLICore/ExecutionContext.h

```diff
@@ -214,6 +215,13 @@ namespace AppInstaller::CLI::Execution
             using value_t = std::vector<std::shared_ptr<Repository::ISource>>;
         };
 
+        template <>
+        struct DataMapping<Data::ARPSnapshot>
+        {
+            // Contains the { Id, Version, Channel }

```

Comment:
>{ Id, Version, Channel } [](start = 28, length = 24)

Do we want a simple struct for this?

Reply:
I would have, but it needs to be a concrete type at this point and I was being lazy.  If we are going to do that I think the best choice would be to make a separate header just for defining these Execution::Data types to not crowd this header.

Reply:
Also, if I were to do so I would need to implement several operators that tuple does for me.  I just don't see the value in that to get named fields.

However, I did do the work to make a new header and move the data part there so that a future item could be added.  I also moved the variant code to language utilities.  Eventually the uses of it that you added for manifests should be moved over to the common code.